utils/cleanstack.go

```diff
@@ -0,0 +1,58 @@
+package utils
+
+import (
+	"github.com/hashicorp/go-multierror"
+)
+
+type CleanJob func() error
+
+// NewCleanStack returns a new stack.
+// It's used to push jobs into it that need to be executed in order, like unmounting disks or removing dirs, and it
+// will run those jobs in the order they were pushed into it to maintain order
+// So you can create a dir, push its removal into the stack, mount something into that dir and push its unmounting into
+// the stack and when cleanup is triggered it will first unmount and then remove the dir
+// Usually its setup inside a function with a defer immediately so it auto cleans if you return from anywhere in the function
+// That way you don't need to track on each return what needs to be cleaned and whatnot
+// cleanup := utils.NewCleanStack()
+// defer func() { err = cleanup.Cleanup(err) }()
+func NewCleanStack() *CleanStack {
+	return &CleanStack{}
+}
+
+// CleanStack is a basic LIFO stack that resizes as needed.
+type CleanStack struct {
+	jobs  []CleanJob
+	count int
```

Comment:
shouldn't count be a function that reads how many items there are in `[]CleanJob`. Maybe I'm not seeing it clearly but I don't see the value of keeping the count of an array which we control

Reply:
mmmh, I guess its because we dont really pop stuff from the jobs but access the index by the count so we can move over it?

So its not really the count I gues, but the index?

Reply:
yeah, I'd call it position or index